### PR TITLE
Update cyclonedx-lib/build.xml to clone openkeystore and build as part of the build ant target

### PR DESCRIPTION
**Pull request by julian55455**
Update cyclonedx-lib/build.xml to clone openkeystore and build as part of the build ant target

Ref: https://github.com/adoptium/temurin-build/issues/3158

### CODE REVIEW

**Comment by julian55455** [comment]
@andrew-m-leonard @zdtsw @smlambert 
This triggers the build though for now openkeystore isn't building successfully on my machine locally, probably the issue is with the resources in my machine or something else

```
ntn@ntn:~/Videos/temurin-build/cyclonedx-lib$ ant build
Buildfile: /home/ntn/Videos/temurin-build/cyclonedx-lib/build.xml
  [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.

dep-checks:

download-cyclonedx:

download-jackson-core:

download-jackson-annotations:

download-json-schema:

download-commons-codec:

download-commons-io:

download-github-package-url:

clone-and-build-openkeystore:
     [exec] Cloning into 'openkeystore'...
     [exec] Buildfile: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/build.xml
     [exec] 
     [exec] build:
     [exec] 
     [exec] _jdktest:
     [exec]      [echo] 
     [exec]      [echo]  ***************************************************************
     [exec]      [echo]     WARNING - This package has only been tested with Java 17!
     [exec]      [echo]  ***************************************************************
     [exec]      [echo] 
     [exec] 
     [exec] _preprocess:
     [exec]     [mkdir] Created dir: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]      [copy] Copying 22 files to /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]     [javac] Compiling 347 source files to /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]     [javac] /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/src/org/webpki/crypto/OkpSupport.java:28: error: cannot find symbol
     [exec]     [javac] import java.security.interfaces.EdECKey;
     [exec]     [javac]                                ^
     [exec]     [javac]   symbol:   class EdECKey
     [exec]     [javac]   location: package java.security.interfaces
     [exec]     [javac] /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/src/org/webpki/crypto/OkpSupport.java:134: error: cannot find symbol
     [exec]     [javac]         if (key instanceof EdECKey) {
     [exec]     [javac]                            ^
     [exec]     [javac]   symbol:   class EdECKey
     [exec]     [javac]   location: class OkpSupport
     [exec]     [javac] /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/src/org/webpki/crypto/OkpSupport.java:136: error: cannot find symbol
     [exec]     [javac]                     ((EdECKey)key).getParams().getName(),
     [exec]     [javac]                       ^
     [exec]     [javac]   symbol:   class EdECKey
     [exec]     [javac]   location: class OkpSupport
     [exec]     [javac] 3 errors
     [exec] 
     [exec] BUILD FAILED
     [exec] /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/build.xml:124: Compile failed; see the compiler error output for details.
     [exec] 
     [exec] Total time: 8 seconds
     [exec] Result: 1

build:

BUILD SUCCESSFUL
Total time: 46 seconds

```

**Comment by zdtsw** [comment]
> EdECKey

will you be able to use a temurin jdk17 docker image and try "ant build" for openkeystore inside of the container?

I have jdk17 locally and it works for me. but if the limitation is it can only be compiled by jdk17, then we need to switch to use jdk17 on our build machine for TemurinGenSBOM to work (if we want to build from source)

**Comment by julian55455** [comment]
> > EdECKey
> 
> will you be able to use a temurin jdk17 docker image and try "ant build" for openkeystore inside of the container?
> 
> I have jdk17 locally and it works for me. but if the limitation is it can only be compiled by jdk17, then we need to switch to use jdk17 on our build machine for TemurinGenSBOM to work (if we want to build from source)

mine too works using jdk17

```
dep-checks:

download-cyclonedx:

download-jackson-core:

download-jackson-annotations:

download-json-schema:

download-commons-codec:

download-commons-io:

download-github-package-url:

clone-and-build-openkeystore:
     [exec] Cloning into 'openkeystore'...
     [exec] Buildfile: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/build.xml
     [exec] 
     [exec] build:
     [exec] 
     [exec] _jdktest:
     [exec] 
     [exec] _preprocess:
     [exec]     [mkdir] Created dir: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]      [copy] Copying 22 files to /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]     [javac] Compiling 347 source files to /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec]     [mkdir] Created dir: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/dist
     [exec]       [jar] Building jar: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/dist/webpki.org-libext-1.00.jar
     [exec]       [jar] Building jar: /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/dist/webpki.org-webutil-1.00.jar
     [exec]     [javac] Compiling 4 source files to /home/ntn/Videos/temurin-build/cyclonedx-lib/openkeystore/library/.tmp
     [exec] 
     [exec] _preprocess:
     [exec] 
     [exec] BUILD SUCCESSFUL
     [exec] Total time: 11 seconds

build:

BUILD SUCCESSFUL
Total time: 1 minute 9 seconds

```

**Comment by karianna** [comment]
@julian55455 same as other PR - needs an Eclipse signed CLA

**Comment by julian55455** [comment]
> @julian55455 same as other PR - needs an Eclipse signed CLA

@karianna I have re-validated the signing a couple of times but builds still fail due to Eclipse CLA on almost every PR I push

**Comment by karianna** [comment]
@julian55455 The email address attached to the commits is (atuhwerajulian2@****.com) - is that the email address that you have the ECA signed for?

**Comment by julian55455** [comment]
> @julian55455 The email address attached to the commits is (atuhwerajulian2@****.com) - is that the email address that you have the ECA signed for?

Yes that the same email I use

**Comment by karianna** [comment]
Hmm will need to contact Eclipse Foundation about that in that case.

File: cyclonedx-lib/build.xml [hunk 1]
```diff
@@ -55,6 +55,16 @@
...
         </target>
 
+        <target name="clone-and-build-openkeystore">
+                <exec dir="./" executable="git">
+                <arg value="clone" />
+                <arg value="https://github.com/cyberphone/openkeystore.git" />
+                </exec>
+                <exec dir="./openkeystore/library" executable="ant">
+                <arg value="build" />
+                </exec>
```
Comment (zdtsw) [inline cyclonedx-lib/build.xml:65]: need to add more step to copy the target into destdir="build/jar"

File: cyclonedx-lib/build.xml [hunk 2]
```diff
@@ -55,6 +55,20 @@
...
+                <copy todir="build/jar">
+                <fileset dir="openkeystore/library/build/jar" includes="*.jar" />
```
Comment (zdtsw) [inline cyclonedx-lib/build.xml]:
```suggestion
                <fileset dir="openkeystore/library/dist" includes="*.jar" />
```

**Review by zdtsw — APPROVED** [review]

**Review by andrew-m-leonard — APPROVED** [review]

**Comment by zdtsw** [comment]
**need to test this out first, once code frozen is uplifted**

**Comment by zdtsw** [comment]
you need rebase this, it is failing with smoketest

**Pull request merged**